app/src/main/res/layout/activity_places_list.xml

```diff
@@ -3,47 +3,57 @@
-                android:fitsSystemWindows="true"
-                android:elevation="0dp"
                 android:background="@color/color_primary"
-                app:title="@string/places_list_title"
+                android:elevation="0dp"
+                android:fitsSystemWindows="true"
                 android:theme="@style/ThemeOverlay.AppCompat.Dark.ActionBar"
+                app:layout_constraintTop_toTopOf="parent"
+                app:layout_constraintBottom_toTopOf="@+id/places_list_filter_button_bar"
                 app:popupTheme="@style/ThemeOverlay.AppCompat.Light"
+                app:title="@string/places_list_title"
                 tools:targetApi="lollipop" />
 
-            <include layout="@layout/filter_button_bar"
+            <include
                 android:id="@+id/places_list_filter_button_bar"
+                layout="@layout/filter_button_bar"
                 android:layout_width="match_parent"
                 android:layout_height="wrap_content"
-                android:layout_below="@+id/places_list_toolbar" />
-
-            <android.support.v7.widget.RecyclerView
-                android:id="@+id/places_list_recycler_view"
-                android:layout_below="@id/places_list_filter_button_bar"
-                android:layout_width="match_parent"
-                android:layout_height="wrap_content">
-            </android.support.v7.widget.RecyclerView>
+                app:layout_constraintBottom_toTopOf="@+id/places_list_recycler_view"
+                app:layout_constraintTop_toBottomOf="@+id/places_list_filter_button_bar" />
+        </android.support.design.widget.AppBarLayout>
 
-            <TextView
-                android:id="@+id/empty_places_list"
-                android:layout_width="match_parent"
-                android:layout_height="match_parent"
-                android:gravity="center"
-                android:visibility="gone"
-                android:text="@string/filter_no_results" />
+        <android.support.v7.widget.RecyclerView
+            android:id="@+id/places_list_recycler_view"
+            app:layout_constraintTop_toBottomOf="@id/places_list_app_bar"
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_behavior="@string/appbar_scrolling_view_behavior"
+            android:layout_width="match_parent"
+            android:layout_height="match_parent">
+        </android.support.v7.widget.RecyclerView>
 
-        </RelativeLayout>
+        <TextView
+            android:id="@+id/empty_places_list"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:gravity="center"
+            android:text="@string/filter_no_results"
+            android:visibility="gone"
+            app:layout_constraintTop_toBottomOf="@+id/places_list_app_bar" />
```

Comment:
This isn't placed correctly - it shows up in the top-left hidden underneath the toolbar.

I'm guessing by the `app:layout_constraintTop_toBottomOf` attribute that you intended to add a `ConstraintLayout` to this layout?

Reply:
Close, I'd removed a `ConstraintLayout`.